src/i_video.c

```diff
@@ -527,27 +527,15 @@ static void ProcessEvent(SDL_Event *ev)
 
 static void I_GetEvent(void)
 {
-    #define NUM_PEEP 32
-    static SDL_Event sdlevents[NUM_PEEP];
-
     I_DelayEvent();
 
     SDL_PumpEvents();
 
-    while (true)
-    {
-        const int num_events = SDL_PeepEvents(sdlevents, NUM_PEEP, SDL_GETEVENT,
-                                              SDL_FIRSTEVENT, SDL_LASTEVENT);
+    SDL_Event sdlevent;
 
-        if (num_events < 1)
-        {
-            break;
-        }
-
-        for (int i = 0; i < num_events; i++)
-        {
-            ProcessEvent(&sdlevents[i]);
-        }
+    while (SDL_PollEvent(&sdlevent))
```

Comment:
This is going to undo the performance gained by this PR: https://github.com/fabiangreffrath/woof/pull/1753

Reply:
Maybe we should wait until SDL3 with this PR?

Reply:
I don't know, but if I can make a request: Please save big changes to the input code until after the next release.

Reply:
Okay, I guess this issue is not that important. The SDL3 branch is the next thing I planned to do after the release.

Reply:
Would you guys please mind having a look at the solution I proposed? This increases the size of an array by one.

Reply:
> Would you guys please mind having a look at the solution I proposed? This increases the size of an array by one.

`event.data3.i = GetTypedChar(&sdlevent->key.keysym, sdlevent + 1);` I don't understand this line, we assume that there is always next event in queue?

While the diff is small, it's going to be hard to remember what's going on here in a couple weeks 😄

Reply:
> `event.data3.i = GetTypedChar(&sdlevent->key.keysym, sdlevent + 1);` I don't understand this line, we assume that there is always next event in queue?

Yes, indeed there always is, because the array that holds the event queue in `I_GetEvent()` is allocated one element larger than the number of events retrieved by `SDL_PeepEvents()` in that same function. So, at this point we just need to check if that "next" element is actually of `SDL_TEXTINPUT` type and then read the typed char fro it. From this point on it's the same as the Chocolate code, but we peep into SDL's event queue at a different point.

> While the diff is small, it's going to be hard to remember what's going on here in a couple weeks 😄

We have comments for this. :wink:

Anyway, I think the solution that you proposed on Discord, i.e. use `I_StartTextInput()`/`I_StopTextInput()` to explicitly enable or disable processing of `SDL_TEXTINPUT` type events and directly feed the typed char into the game's event queue, is preferable.

Reply:
> Anyway, I think the solution that you proposed on Discord, i.e. use `I_StartTextInput()`/`I_StopTextInput()` to explicitly enable or disable processing of `SDL_TEXTINPUT` type events and directly feed the typed char into the game's event queue, is preferable.

Done here: 6f6d2eb